experimental/cmd/mcp9808/main.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// mcp9808 communicates with an mcp9808 sensor reading ambient temperature.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/experimental/devices/mcp9808"
+	"periph.io/x/periph/host"
+)
+
+func mainImpl() error {
+	if _, err := host.Init(); err != nil {
+		return err
+	}
+	address := flag.Int("address", 0x18, "I²C address")
+	i2cbus := flag.String("bus", "", "I²C bus (/dev/i2c-1)")
+
+	flag.Parse()
+
+	fmt.Println("Starting MCP9808 Temperature Sensor")
+	if _, err := host.Init(); err != nil {
+		return err
+	}
+
+	// Open default I²C bus.
+	bus, err := i2creg.Open(*i2cbus)
+	if err != nil {
+		return fmt.Errorf("failed to open I²C: %v", err)
+	}
+	defer bus.Close()
+
+	// Create a new temperature sensor a sense with default options.
+	sensor, err := mcp9808.New(bus, &mcp9808.Opts{Address: *address})
+	if err != nil {
+		return fmt.Errorf("failed to open new sensor: %v", err)
+	}
+
+	// Read values from sensor every second.
+	everySecond := time.NewTicker(time.Second).C
+	var halt = make(chan os.Signal)
+	signal.Notify(halt, syscall.SIGTERM)
+	signal.Notify(halt, syscall.SIGINT)
+
+	fmt.Println("ctrl+c to exit")
+	for {
+		select {
+		case <-everySecond:
+			t, err := sensor.Sense()
```

Comment:
One interesting thing here: the first reading will be after 1s, instead of immediately at program startup.

Reply:
Personally I normally do stuff on the tick of the clock. So I don't have any good clear solutions ready to go.